### PR TITLE
[Inference API] Add Docs for AlibabaCloud AI Search Support for the Inference API

### PR DESCRIPTION
**Pull request by weizijun**
Add docs in support of AlibabaCloud AI Search support in the Inference API: https://github.com/elastic/elasticsearch/pull/111181

### CODE REVIEW

**Comment by weizijun** [comment]
hi, @davidkyle, this is the PR document of AlibabaCloud AI Search Inference. Can you help review it?

**Comment by davidkyle** [comment]
Thanks @weizijun, I've assigned to @szabosteve who writes the docs

File: docs/reference/inference/service-alibabacloud-ai-search.asciidoc [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+The service_id available for the `text_embedding` task are:
+
+* `ops-text-embedding-001`
+* `ops-text-embedding-zh-001`
+* `ops-text-embedding-en-001`
+* `ops-text-embedding-002`
+
+You can find the supported `text_embedding` service_id at https://help.aliyun.com/zh/open-search/search-platform/developer-reference/text-embedding-api-details[embedding detail]
+
```
Comment (szabosteve) [inline docs/reference/inference/service-alibabacloud-ai-search.asciidoc:65]:
```suggestion
For the supported `text_embedding` service_id, refer to the https://help.aliyun.com/zh/open-search/search-platform/developer-reference/text-embedding-api-details[documentation].

```

File: docs/reference/inference/service-alibabacloud-ai-search.asciidoc [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+for the `sparse_embedding` task are:
```
Comment (szabosteve) [inline docs/reference/inference/service-alibabacloud-ai-search.asciidoc:66]: Nit.

```suggestion
The service_id available for the `sparse_embedding` task is:
```

File: docs/reference/inference/service-alibabacloud-ai-search.asciidoc [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+* `ops-text-sparse-embedding-001`
+
+You can find the supported `sparse_embedding` service_id at https://help.aliyun.com/zh/open-search/search-platform/developer-reference/text-sparse-embedding-api-details[sparse embedding detail]
```
Comment (szabosteve) [inline docs/reference/inference/service-alibabacloud-ai-search.asciidoc:70]:
```suggestion
For the supported `sparse_embedding` service_id, refer to the https://help.aliyun.com/zh/open-search/search-platform/developer-reference/text-sparse-embedding-api-details[documentation].
```

File: docs/reference/inference/service-alibabacloud-ai-search.asciidoc [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+for the `rerank` task are:
```
Comment (szabosteve) [inline docs/reference/inference/service-alibabacloud-ai-search.asciidoc:72]:
```suggestion
The service_id available for the `rerank` task is:
```

File: docs/reference/inference/service-alibabacloud-ai-search.asciidoc [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+* `ops-bge-reranker-larger`
+
+You can find the supported `rerank` service_id at https://help.aliyun.com/zh/open-search/search-platform/developer-reference/ranker-api-details[rerank detail]
```
Comment (szabosteve) [inline docs/reference/inference/service-alibabacloud-ai-search.asciidoc:76]:
```suggestion
For the supported `rerank` service_id, refer to the https://help.aliyun.com/zh/open-search/search-platform/developer-reference/ranker-api-details[documentation].
```

File: docs/reference/inference/service-alibabacloud-ai-search.asciidoc [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+
+`host`:::
+(Required, string)
+The name of the host address use for the {infer} task. You can find the host address at https://opensearch.console.aliyun.com/cn-shanghai/rag/api-key[API keys section]
```
Comment (szabosteve) [inline docs/reference/inference/service-alibabacloud-ai-search.asciidoc:80]:
```suggestion
The name of the host address used for the {infer} task. You can find the host address at https://opensearch.console.aliyun.com/cn-shanghai/rag/api-key[ the API keys section] of the documentation.
```

File: docs/reference/inference/service-alibabacloud-ai-search.asciidoc [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+
+`workspace`:::
+(Required, string)
+The name of the workspace use for the {infer} task.
```
Comment (szabosteve) [inline docs/reference/inference/service-alibabacloud-ai-search.asciidoc:84]:
```suggestion
The name of the workspace used for the {infer} task.
```

File: docs/reference/inference/service-alibabacloud-ai-search.asciidoc [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
++
+.`task_settings` for the `text_embedding` task type
+[%collapsible%closed]
+=====
+`input_type`:::
+(Optional, string)
+Specifies the type of input passed to the model.
+Valid values are:
+* `ingest`: use it for storing document embeddings in a vector database.
+* `search`: use it for storing embeddings of search queries run against a vector database to find relevant documents.
```
Comment (szabosteve) [inline docs/reference/inference/service-alibabacloud-ai-search.asciidoc:108]:
```suggestion
* `ingest`: for storing document embeddings in a vector database.
* `search`: for storing embeddings of search queries run against a vector database to find relevant documents.
```

File: docs/reference/inference/service-alibabacloud-ai-search.asciidoc [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
++
+
```
Comment (szabosteve) [inline docs/reference/inference/service-alibabacloud-ai-search.asciidoc:110]:
```suggestion
```

File: docs/reference/inference/service-alibabacloud-ai-search.asciidoc [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+.`task_settings` for the `sparse_embedding` task type
+[%collapsible%closed]
+=====
+`input_type`:::
+(Optional, string)
+Specifies the type of input passed to the model.
+Valid values are:
+* `ingest`: use it for storing document embeddings in a vector database.
+* `search`: use it for storing embeddings of search queries run against a vector database to find relevant documents.
++
```
Comment (szabosteve) [inline docs/reference/inference/service-alibabacloud-ai-search.asciidoc:122]:
```suggestion
```

File: docs/reference/inference/service-alibabacloud-ai-search.asciidoc [hunk 11]
```diff
@@ -0,0 +1,193 @@
...
+.`task_settings` for the `sparse_embedding` task type
+[%collapsible%closed]
+=====
+`input_type`:::
+(Optional, string)
+Specifies the type of input passed to the model.
+Valid values are:
+* `ingest`: use it for storing document embeddings in a vector database.
+* `search`: use it for storing embeddings of search queries run against a vector database to find relevant documents.
```
Comment (szabosteve) [inline docs/reference/inference/service-alibabacloud-ai-search.asciidoc:121]:
```suggestion
* `ingest`: for storing document embeddings in a vector database.
* `search`: for storing embeddings of search queries run against a vector database to find relevant documents.
```

File: docs/reference/inference/service-alibabacloud-ai-search.asciidoc [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
++
+
+`return_token`:::
+(Optional, boolean)
+Specify Whether to return the token name, the default value is false, it means that it will return the token id.
```
Comment (szabosteve) [inline docs/reference/inference/service-alibabacloud-ai-search.asciidoc:126]:
```suggestion
If `true`, the token name will be returned in the response. Defaults to `false` which means only the token ID will be returned in the response.
```

File: docs/reference/inference/service-alibabacloud-ai-search.asciidoc [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+
```
Comment (szabosteve) [inline docs/reference/inference/service-alibabacloud-ai-search.asciidoc:127]:
```suggestion
```

File: docs/reference/inference/service-alibabacloud-ai-search.asciidoc [hunk 14]
```diff
@@ -0,0 +1,193 @@
...
+=====
++
+.`task_settings` for the `rerank` task type
+[%collapsible%closed]
+=====
+
+There are no `task_settings` available for the `rerank` task type.
+
+=====
```
Comment (szabosteve) [inline docs/reference/inference/service-alibabacloud-ai-search.asciidoc:136]: I'D suggest removing this section if there is no content for it.

```suggestion
```

File: docs/reference/inference/service-alibabacloud-ai-search.asciidoc [hunk 15]
```diff
@@ -0,0 +1,193 @@
...
+
+[discrete]
+[[inference-example-alibabacloud-ai-search]]
+==== AlibabaCloud AI Search service example
```
Comment (szabosteve) [inline docs/reference/inference/service-alibabacloud-ai-search.asciidoc:140]: Nit.

```suggestion
==== AlibabaCloud AI Search service examples
```

File: docs/reference/tab-widgets/inference-api/infer-api-requirements.asciidoc [hunk 16]
```diff
@@ -52,3 +52,9 @@ You can apply for access to Azure OpenAI by completing the form at https://aka.m
 * A pair of access and secret keys used to access Amazon Bedrock
 
 // end::amazon-bedrock[]
+
+// tag::alibabacloud-ai-search[]
+* An AlibabaCloud Account with https://console.aliyun.com[AlibabaCloud] access
+* An API key generated for your account from https://opensearch.console.aliyun.com/cn-shanghai/rag/api-key[API keys section]
```
Comment (szabosteve) [inline docs/reference/tab-widgets/inference-api/infer-api-requirements.asciidoc:58]:
```suggestion
* An API key generated for your account from the https://opensearch.console.aliyun.com/cn-shanghai/rag/api-key[API keys section]
```

**Review by szabosteve — COMMENTED** [review]
Great documentation! Thank you for drafting this! I left a couple of suggestions, mostly nits.

**Comment by weizijun** [comment]
> Great documentation! Thank you for drafting this! I left a couple of suggestions, mostly nits.

Sorry for my poor English, I have submitted the changes directly on GitHub, thank you!

**Comment by szabosteve** [comment]
> Sorry for my poor English, I have submitted the changes directly on GitHub, thank you!
Your English is fine! Thank you for this huge docs contribution.
I checked the pages out locally and they build fine. I pushed some indentation fixes via [c591269](https://github.com/elastic/elasticsearch/pull/112273/commits/c591269f21a936ce4318d14ef7e5c4fd3b6ba8cc). Hope you don't mind.

**Review by szabosteve — APPROVED** [review]
LGTM! Thanks for writing the docs!

**Comment by weizijun** [comment]
> I checked the pages out locally and they build fine. I pushed some indentation fixes via [c591269](https://github.com/elastic/elasticsearch/pull/112273/commits/c591269f21a936ce4318d14ef7e5c4fd3b6ba8cc). Hope you don't mind.

It doesn't matter, thanks!

**Comment by szabosteve** [comment]
@elasticsearchmachine test this please

**Comment by szabosteve** [comment]
@elasticsearchmachine merge upstream

**Pull request merged**